app/views/posts/index.html.erb

```diff
@@ -0,0 +1,31 @@
+<p id="notice"><%= notice %></p>
+
+<h1>Listing Posts</h1>
+
+<table>
+  <thead>
+    <tr>
+      <th>Title</th>
+      <th>Contents</th>
+      <th>Author</th>
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @posts.each do |post| %>
+      <tr>
+        <td><%= post.title %></td>
+        <td><%= post.contents %></td>
+        <td><%= User.find(post.user_id).username %></td>
```

Comment:
If the foreign keys are set up right you should be able to do `post.user.username`.

What I meant when I said inefficient is that you are doing one query for every post, there is a way to tell rails to combine those all into one query but we can worry about that later.